governator-core/src/main/java/com/netflix/governator/LifecycleManager.java

```diff
@@ -60,17 +62,19 @@ public synchronized void notifyStarted() {
     public synchronized void notifyStartFailed(Throwable t) {
         if (state.compareAndSet(State.Starting, State.Stopped)) {
             failureReason = t;
-            for (LifecycleListener listener : listeners) {
-                listener.onStopped(t);
+            LifecycleListener[] asArray = listeners.toArray(new LifecycleListener[0]);
```

Comment:
nit: I think the code would read better if we use an iterator instead of dealing with raw arrays. 

```java
Iterator<String> iter = new LinkedList<>(listeners).descendingIterator();
while (iter.hasNext()) {
    iter.next().onStopped();
}
```